qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/ClusterStateIT.java

```diff
@@ -8,30 +8,369 @@
 
 package org.opensearch.upgrades;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.opensearch.client.Request;
 import org.opensearch.client.Response;
+import org.opensearch.cluster.metadata.IndexTemplateMetadata;
+import org.opensearch.common.settings.Settings;
 
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
 
+@SuppressWarnings("unchecked")
 public class ClusterStateIT extends AbstractRollingTestCase{
+    private static final String templateName = "my_template";
+    private static final String indexName = "test_cluster_state";
+    private static final String componentTemplateName = "test_component_template";
+
+    // Add different types of metadata to Cluster State to ensure the new versions understand these metadata types in future versions
+    public void testUpgradeWithRemotePublicationEnabled() throws Exception {
+        if (CLUSTER_TYPE == ClusterType.OLD) {
+            // verify that Remote Publication is Enabled on the cluster
+            verifyRemotePublicationEnabled();
+            // Create Index
+            Settings indexSettings = Settings.builder()
+                .put("index.number_of_shards", 1)
+                .put("index.number_of_replicas", 1)
+                .put("index.refresh_interval", "5s").build();
+            String indexMappings = "\"properties\": {\"name\": {\"type\":\"text\"}}";
+            String aliases = "\"main\":{\"is_write_index\":true}";
+            createIndex(indexName, indexSettings, indexMappings, aliases);
+            // Create Index Template
+            createIndexTemplate();
+            // Create Component Template
+            createComponentTemplate();
+            // Create Composable Template
+            createComposableTemplate();
+            // Add Settings
+            addPersistentSettings();
+            addTransientSettings();
+
+            Request request = new Request("GET", "_cluster/state");
+            Response response = client().performRequest(request);
+            assertOK(response);
+
+            verifyIndexInClusterState(response);
+            verifyTemplateMetadataInClusterState(response);
+            verifyComponentTemplateInClusterState(response);
+            verifyComposableTemplateInClusterState(response);
+            verifySettingsInClusterState();
+        } else {
+            verifyRemotePublicationEnabled();
+
+            Request request = new Request("GET", "_cluster/state");
+            Response response = client().performRequest(request);
+            assertOK(response);
+
+            verifyIndexInClusterState(response);
+            verifyTemplateMetadataInClusterState(response);
+            verifyComponentTemplateInClusterState(response);
+            verifyComposableTemplateInClusterState(response);
+//            Settings are required to wiped after Tests, so Settings are not present after rolling upgrade starts
```

Comment:
Which settings are wiped? But remote publication enabled setting is being verified above.

Reply:
Added few persistent and transient settings in the old version cluster, but we clean up the settings after test is run to run other tests.

https://github.com/opensearch-project/OpenSearch/blob/08dc3bb9cb8a4cd01ab32d435e80a41542bbc866/test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java#L583-L585